crates/db_schema/src/source/custom_emoji.rs

```diff
@@ -0,0 +1,34 @@
+use crate::newtypes::LocalSiteId;
+#[cfg(feature = "full")]
+use crate::schema::custom_emoji;
+use serde::{Deserialize, Serialize};
+
+#[derive(PartialEq, Eq, Debug, Clone, Serialize, Deserialize)]
+#[cfg_attr(feature = "full", derive(Queryable, Associations, Identifiable))]
+#[cfg_attr(feature = "full", diesel(table_name = custom_emoji))]
+#[cfg_attr(
+  feature = "full",
+  diesel(belongs_to(crate::source::local_site::LocalSite))
+)]
+pub struct CustomEmoji {
+  pub id: i32,
+  pub local_site_id: LocalSiteId,
+  pub shortcode: String,
+  pub image_url: String,
+  pub alt_text: String,
+  pub category: String,
+  pub published: chrono::NaiveDateTime,
+  pub updated: Option<chrono::NaiveDateTime>,
+}
+
+#[derive(Clone)]
+#[cfg_attr(feature = "full", derive(Insertable, AsChangeset))]
+#[cfg_attr(feature = "full", diesel(table_name = custom_emoji))]
+pub struct CustomEmojiForm {
+  pub local_site_id: LocalSiteId,
+  pub shortcode: String,
+  pub image_url: String,
```

Comment:
This would probably be better as a `DbUrl`, check out the url field for `post.rs`